nncf/quantization/algo.py

```diff
@@ -416,9 +411,35 @@ def _quantize_activations(self, target_model: NNCFNetwork) -> List[InsertionComm
                     err_msg += str(ia_op_exec_context)
                     raise RuntimeError(err_msg)
 
+                final_quantizer_config_setup[insertion_info] = quantizer_config
+
+            finalized_proposal = quantization_proposal.finalize(final_quantizer_config_setup)
+            final_quantizer_setup = prop_graph_solver.get_final_quantizer_setup(finalized_proposal)
```

Comment:
2) The quantizer propagation algorithm has been modified to work in two stages. This is related to the new quantizer configuration merging algorithm. Basically the first step of the propagation results in a proposal - a set of the points at which a quantizer should be inserted in order to quantize all operations specified in HW config with a supported configuration, and, for each point, a list of possible quantizer configurations out of which one and only one should be chosen. The choice could be either affected by utilizing the NNCF config options/presets, or it could be delegated to a specialized algorithm, such as HAWQ in case multiple bitwidths are present in the quantizer configuration list, or an RL algorithm. 

Once the final quantizer configurations are chosen, an extra step is necessary to merge the quantizers that are redundant - e.g. if one and the same final quantizer configuration has been chosen both for the global quantizer of all downstream branches and for the local quantizer on a branch.